components/ILIAS/Saml/classes/class.ilSamlSettingsGUI.php

```diff
@@ -497,17 +497,24 @@ private function showUserAttributeMappingForm(?ilPropertyFormGUI $form = null):
 
     private function getSettingsForm(array $values = []): StandardForm
     {
-        return $this->ui_factory->input()->container()->form()->standard(
+        $access = $this->rbac_system->checkAccess(self::PERMISSION_WRITE, $this->ref_id);
+        $form = $this->ui_factory->input()->container()->form()->standard(
             $this->ctrl->getFormAction($this, self::CMD_SAVE_SETTINGS),
```

Comment:
@mBeym We should change the action as well (to the one relevant for displaying the form) if no `write` permission is granted.